docs: warn about modules with side-effects (#19592)
nuxt/nuxt: docs/2.guide/1.concepts/3.rendering.md

For more examples about writing Vue code without hydration mismatch, see [the Vue docs](https://vuejs.org/guide/scaling-up/ssr.html#hydration-mismatch).
::

::alert{type=warning}

> When importing libraries with side-effects relying on browser API, make sure the component importing it is only called client-side. Bundlers does no treeshake imports of modules containing side-effects.

Reviewer: ```suggestion
When importing a library with side-effects that relies on browser APIs, make sure the component importing it is only called client-side. Bundlers do not treeshake imports of modules containing side-effects.
```

> When importing a library with side-effects that relies on browser APIs, make sure the component importing it is only called client-side. Bundlers do not treeshake imports of modules containing side-effects.

Reviewer: ```suggestion
When importing a library that relies on browser APIs and has side effects, make sure the component importing it is only called client-side. Bundlers do not treeshake imports of modules containing side-effects.
```
